### DISCUSSION TITLE
Sonoff SV with DS18x20 Temperature Sensor

### DISCUSSION THREAD

**Discussion by lysiasthefox** [arendst/Tasmota#12971] [Q&A] [answered]
I am using a Sonoff SV (running Tasmota 9.5.0) to control a radiator water inlet (12V valve). I have this controlled using a toggle button on GPIO 5(2) and via OpenHAB. I intend to monitor the water temperature passing through this valve using a DS18B20 and have connected this to GPIO14/VCC/GND and set the Tasmota Module Configuration accordingly, GPIO14 to DS18x20. I had expected to then see the DS18x20 added to the telemetry and perhaps a temperature reading on the Tasmota main menu, but there seems to be no change.  I would appreciate any help or tips on what I may have missed in my setup. Thank you! Nate.

**Comment by ascillato** [comment, marked as answer]
For that, please update to *_Tasmota-sensors.bin_*

Go to the console and type:

```lua
OTAurl http://ota.tasmota.com/tasmota/tasmota-sensors.bin.gz
Upgrade 1
```

**Reply by lysiasthefox** [reply]
Thank you for the response. After updating to Version 9.5.0.7(sensors)-2_7_4_9(2021-08-24T21:00:36) - it appears to have the same result. I have set the module configuration as GPIO14 to DS18x20, but still no sign of a temperature sensor. Thanks again.

**Reply by lysiasthefox** [reply]
Weirdly, enabling the internal pull-up using the console command SetOption74 1 seemed to get things working - thanks again for your response and solution.

**Reply by barbudor** [reply]
One-Wire requires a pull-up resistor on the data line. If you haven't installed one externally, indeed you need to enable the internal one.
It is always recommended to read the datasheets of parts that you are using: https://datasheets.maximintegrated.com/en/ds/DS18B20.pdf
Or at minimum the Tasmota documentation: https://tasmota.github.io/docs/DS18x20/

Note that tasmota-sensor.bin is NOT required for DS18x20 which is already part of tasmota.bin as per the doc [BUILD.md](https://github.com/arendst/Tasmota/blob/development/BUILDS.md)
If it was not included, you wouldn't have been able to assign a GPIO to DS18x20.

**Reply by lysiasthefox** [reply]
Thanks barbudor, I was familiar with the required pull-up (I'm an electrical engineer). I bought off ebay and it suggested that the modules were Rpi/Arduino ready boasting a surface mount resistor (I assume 4k7) but in desperation I decided to quickly test with the internal pull-up.  I bought a number of these wanting to measure two temps on each SV, so I'll check the others and if they too have faulty pull-ups, I'll add them manually (and turn off internal pull-up).  I'll revert back to the standard tasmota.bin - Thanks again

**Reply by sfromis** [reply]
With one DS18B20 sensor (and not too long wires), the weak internal pullup usually does the job just fine.
